src/main/java/frc/robot/util/SwerveModuleAcceleration.java

```diff
@@ -0,0 +1,79 @@
+package frc.robot.util;
+
+import java.util.Objects;
+
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+
+public class SwerveModuleAcceleration implements Comparable<SwerveModuleAcceleration> {
+  public double accelMetersPerSecondSquared = 0;
+  private static SwerveModuleState[] states = new SwerveModuleState[4];
```

Comment:
This should be a non-static, non-array state. You should have two here, `m_previousState` and `m_currentState`. You also need a `m_previousTime` field to properly calculate the acceleration.

Reply:
Unless we want to keep this as a POD class, then we should remove this entirely, and add a `public final Rotation2d m_angle` field

Reply:
m_previousTime has been replaced by including a deltaTime param in SMA.calculate()
m_previousState and m_currState have been added in replacement of the list
m_angle is omitted as module rotational accel is never used in SOK.calculateModulesPositionAccelX or Y